e2e_test/source/basic/kafka.slt

```diff
@@ -326,6 +326,15 @@ create source s23 (bytea_example bytea) with (
   scan.startup.mode = 'earliest'
 ) row format json
 
+statement ok
+create source s24 (id bytea) with (
+  connector = 'kafka',
+  topic = 'kafka_source_format_bytes',
+  type = 'append-only',
```

Comment:
no need to specify `type` here, when you create a source, it is indicates to be append-only
ref https://risingwave.dev for details